### PR TITLE
New method to allow appending to csv without reopening connection

### PR DESCRIPTION
**Pull request by rach**
The goal of this method is to be able to consume a single generator in a different place.

### CODE REVIEW

**Comment by ihat** [comment]
@rach Test?

File: fireflower/targets.py [hunk 1]
```diff
@@ -18,6 +18,13 @@
     'S3CSVTarget'
 ]
 
+def coroutine(f):
+    def wrapper(*args, **kw):
+        c = f(*args, **kw)
+        c.send(None)    # This is the same as calling ``next()``,
```
Comment (kaloramik) [inline fireflower/targets.py]: not sure what this send is for.. can you document?

File: fireflower/targets.py [hunk 2]
```diff
@@ -153,6 +160,28 @@ def write_values(csv_writer, values, header=None):
         for v in values:
             csv_writer.writerow(v)
 
+    @coroutine
+    def append_csv_tuples(self, header_tuple=None):
```
Comment (kaloramik) [inline fireflower/targets.py]: should tuples be here?

**Comment by b11z** [comment]
@kaloramik could you take a 2nd look, we switched it up

File: tests/targets_tests.py [hunk 3]
```diff
@@ -65,6 +65,23 @@ def test_s3_csv(self, compressed, stream):
...
+    def test_csv_stream(self, compressed):
+        conn = boto.connect_s3()
+        conn.create_bucket('some_bucket')
+
+        target = S3CSVTarget('s3://some_bucket/some_file.csv.gz', compressed=compressed)
+        with target.open_csv_stream() as f:
+            f.write_tuple(('a','b'))
+            f.write_tuple((3,4))
+
+        read_result = target.read_csv(engine='python')
```
Comment (kaloramik) [inline tests/targets_tests.py]: why is this python?

File: fireflower/targets.py [hunk 4]
```diff
@@ -18,6 +18,13 @@
     'S3CSVTarget'
 ]
 
+def coroutine(f):
```
Comment (kaloramik) [inline fireflower/targets.py]: is this used?

**Pull request merged**